src/openmc_heat_driver.cpp

```diff
@@ -180,16 +181,20 @@ void OpenmcHeatDriver::init_mappings()
 
 void OpenmcHeatDriver::init_tallies()
 {
+  using gsl::index;
+  using gsl::narrow_cast;
+
   if (openmc_driver_->active()) {
-    // Build vector of material indices to construct tallies; tallies are only
+    // Build vector of cell instances to construct tallies; tallies are only
     // used in the solid regions
-    std::vector<int32_t> mats;
+    std::vector<openmc::CellInstance> cells;
     for (gsl::index c = 0; c < n_solid_cells_; ++c) {
       const auto& cell = openmc_driver_->cells_[c];
-      mats.push_back(cell.material_index_);
+      cells.push_back(
+        {narrow_cast<index>(cell.index_), narrow_cast<index>(cell.instance_)});
```

Comment:
It took me a while to figure out why this cast was necessary.  I finally noticed we have both `enrico::CellInstance` and `openmc::CellInstance`.  Is there any possibility (maybe in a future PR) that we could completely replace the former with the latter everywhere in ENRICO?

Reply:
Yes, it's a possibility. Right now the ENRICO version provides methods for asking about temperature/density, but we could move those into OpenMC (where arguably they belong). I'll create an issue to track this.